### ISSUE TITLE
flog.rb undefined method `sexp_type'

### ISSUE THREAD

**Issue by prtngn** [whitesmith/rubycritic#412]
Hi. Rubycritic don't work.
RUBY VERSION
   ruby 3.1.2p20

BUNDLED WITH
   2.3.14

rubycritic (4.7.0)

```
➜  telegram_sender git:(master) ✗ bundle exec rubycritic
running flay smells

running flog smells
........................./Users/prtngn/.rvm/gems/ruby-3.1.2@telegram_sender/gems/flog-4.6.5/lib/flog.rb:368:in `process_block_pass': undefined method `sexp_type' for nil:NilClass (NoMethodError)

    case arg.sexp_type
            ^^^^^^^^^^
        from /Users/prtngn/.rvm/gems/ruby-3.1.2@telegram_sender/gems/sexp_processor-4.16.1/lib/sexp_processor.rb:269:in `block (2 levels) in process'
        from /Users/prtngn/.rvm/gems/ruby-3.1.2@telegram_sender/gems/sexp_processor-4.16.1/lib/sexp_processor.rb:324:in `error_handler'
        from /Users/prtngn/.rvm/gems/ruby-3.1.2@telegram_sender/gems/sexp_processor-4.16.1/lib/sexp_processor.rb:268:in `block in process'
        from /Users/prtngn/.rvm/gems/ruby-3.1.2@telegram_sender/gems/sexp_processor-4.16.1/lib/sexp_processor.rb:388:in `in_context'
        from /Users/prtngn/.rvm/gems/ruby-3.1.2@telegram_sender/gems/sexp_processor-4.16.1/lib/sexp_processor.rb:245:in `process'
        from /Users/prtngn/.rvm/gems/ruby-3.1.2@telegram_sender/gems/sexp_processor-4.16.1/lib/sexp_processor.rb:692:in `process_until_empty'
        from /Users/prtngn/.rvm/gems/ruby-3.1.2@telegram_sender/gems/flog-4.6.5/lib/flog.rb:398:in `block in process_call'
        from /Users/prtngn/.rvm/gems/ruby-3.1.2@telegram_sender/gems/flog-4.6.5/lib/flog.rb:268:in `penalize_by'
        from /Users/prtngn/.rvm/gems/ruby-3.1.2@telegram_sender/gems/flog-4.6.5/lib/flog.rb:397:in `process_call'
        from /Users/prtngn/.rvm/gems/ruby-3.1.2@telegram_sender/gems/sexp_processor-4.16.1/lib/sexp_processor.rb:269:in `block (2 levels) in process'
        from /Users/prtngn/.rvm/gems/ruby-3.1.2@telegram_sender/gems/sexp_processor-4.16.1/lib/sexp_processor.rb:324:in `error_handler'
        from /Users/prtngn/.rvm/gems/ruby-3.1.2@telegram_sender/gems/sexp_processor-4.16.1/lib/sexp_processor.rb:268:in `block in process'
        from /Users/prtngn/.rvm/gems/ruby-3.1.2@telegram_sender/gems/sexp_processor-4.16.1/lib/sexp_processor.rb:388:in `in_context'
        from /Users/prtngn/.rvm/gems/ruby-3.1.2@telegram_sender/gems/sexp_processor-4.16.1/lib/sexp_processor.rb:245:in `process'
        from /Users/prtngn/.rvm/gems/ruby-3.1.2@telegram_sender/gems/sexp_processor-4.16.1/lib/sexp_processor.rb:692:in `process_until_empty'
        from /Users/prtngn/.rvm/gems/ruby-3.1.2@telegram_sender/gems/sexp_processor-4.16.1/lib/sexp_processor.rb:646:in `block in process_defs'
        from /Users/prtngn/.rvm/gems/ruby-3.1.2@telegram_sender/gems/sexp_processor-4.16.1/lib/sexp_processor.rb:553:in `in_method'
        from /Users/prtngn/.rvm/gems/ruby-3.1.2@telegram_sender/gems/sexp_processor-4.16.1/lib/sexp_processor.rb:642:in `process_defs'
        from /Users/prtngn/.rvm/gems/ruby-3.1.2@telegram_sender/gems/sexp_processor-4.16.1/lib/sexp_processor.rb:269:in `block (2 levels) in process'
        from /Users/prtngn/.rvm/gems/ruby-3.1.2@telegram_sender/gems/sexp_processor-4.16.1/lib/sexp_processor.rb:324:in `error_handler'
        from /Users/prtngn/.rvm/gems/ruby-3.1.2@telegram_sender/gems/sexp_processor-4.16.1/lib/sexp_processor.rb:268:in `block in process'
        from /Users/prtngn/.rvm/gems/ruby-3.1.2@telegram_sender/gems/sexp_processor-4.16.1/lib/sexp_processor.rb:388:in `in_context'
        from /Users/prtngn/.rvm/gems/ruby-3.1.2@telegram_sender/gems/sexp_processor-4.16.1/lib/sexp_processor.rb:245:in `process'
        from /Users/prtngn/.rvm/gems/ruby-3.1.2@telegram_sender/gems/sexp_processor-4.16.1/lib/sexp_processor.rb:692:in `process_until_empty'
        from /Users/prtngn/.rvm/gems/ruby-3.1.2@telegram_sender/gems/flog-4.6.5/lib/flog.rb:436:in `block in process_class'
        from /Users/prtngn/.rvm/gems/ruby-3.1.2@telegram_sender/gems/sexp_processor-4.16.1/lib/sexp_processor.rb:607:in `block in process_class'
        from /Users/prtngn/.rvm/gems/ruby-3.1.2@telegram_sender/gems/sexp_processor-4.16.1/lib/sexp_processor.rb:539:in `block in in_klass'
        from /Users/prtngn/.rvm/gems/ruby-3.1.2@telegram_sender/gems/sexp_processor-4.16.1/lib/sexp_processor.rb:710:in `with_new_method_stack'
        from /Users/prtngn/.rvm/gems/ruby-3.1.2@telegram_sender/gems/sexp_processor-4.16.1/lib/sexp_processor.rb:538:in `in_klass'
        from /Users/prtngn/.rvm/gems/ruby-3.1.2@telegram_sender/gems/sexp_processor-4.16.1/lib/sexp_processor.rb:605:in `process_class'
        from /Users/prtngn/.rvm/gems/ruby-3.1.2@telegram_sender/gems/flog-4.6.5/lib/flog.rb:432:in `process_class'
        from /Users/prtngn/.rvm/gems/ruby-3.1.2@telegram_sender/gems/sexp_processor-4.16.1/lib/sexp_processor.rb:269:in `block (2 levels) in process'
        from /Users/prtngn/.rvm/gems/ruby-3.1.2@telegram_sender/gems/sexp_processor-4.16.1/lib/sexp_processor.rb:324:in `error_handler'
        from /Users/prtngn/.rvm/gems/ruby-3.1.2@telegram_sender/gems/sexp_processor-4.16.1/lib/sexp_processor.rb:268:in `block in process'
        from /Users/prtngn/.rvm/gems/ruby-3.1.2@telegram_sender/gems/sexp_processor-4.16.1/lib/sexp_processor.rb:388:in `in_context'
        from /Users/prtngn/.rvm/gems/ruby-3.1.2@telegram_sender/gems/sexp_processor-4.16.1/lib/sexp_processor.rb:245:in `process'
        from /Users/prtngn/.rvm/gems/ruby-3.1.2@telegram_sender/gems/flog-4.6.5/lib/flog.rb:230:in `flog_ruby!'
        from /Users/prtngn/.rvm/gems/ruby-3.1.2@telegram_sender/gems/flog-4.6.5/lib/flog.rb:195:in `flog_ruby'
        from /Users/prtngn/.rvm/gems/ruby-3.1.2@telegram_sender/gems/flog-4.6.5/lib/flog.rb:183:in `block in flog'
        from /Users/prtngn/.rvm/gems/ruby-3.1.2@telegram_sender/gems/flog-4.6.5/lib/flog.rb:178:in `each'
        from /Users/prtngn/.rvm/gems/ruby-3.1.2@telegram_sender/gems/flog-4.6.5/lib/flog.rb:178:in `flog'
        from /Users/prtngn/.rvm/gems/ruby-3.1.2@telegram_sender/gems/rubycritic-4.7.0/lib/rubycritic/analysers/smells/flog.rb:35:in `add_smells_to'
        from /Users/prtngn/.rvm/gems/ruby-3.1.2@telegram_sender/gems/rubycritic-4.7.0/lib/rubycritic/analysers/smells/flog.rb:21:in `block in run'
        from /Users/prtngn/.rvm/gems/ruby-3.1.2@telegram_sender/gems/rubycritic-4.7.0/lib/rubycritic/core/analysed_modules_collection.rb:32:in `each'
        from /Users/prtngn/.rvm/gems/ruby-3.1.2@telegram_sender/gems/rubycritic-4.7.0/lib/rubycritic/core/analysed_modules_collection.rb:32:in `each'
        from /Users/prtngn/.rvm/gems/ruby-3.1.2@telegram_sender/gems/rubycritic-4.7.0/lib/rubycritic/analysers/smells/flog.rb:20:in `run'
        from /Users/prtngn/.rvm/gems/ruby-3.1.2@telegram_sender/gems/rubycritic-4.7.0/lib/rubycritic/analysers_runner.rb:32:in `block in run'
        from /Users/prtngn/.rvm/gems/ruby-3.1.2@telegram_sender/gems/rubycritic-4.7.0/lib/rubycritic/analysers_runner.rb:29:in `each'
        from /Users/prtngn/.rvm/gems/ruby-3.1.2@telegram_sender/gems/rubycritic-4.7.0/lib/rubycritic/analysers_runner.rb:29:in `run'
        from /Users/prtngn/.rvm/gems/ruby-3.1.2@telegram_sender/gems/rubycritic-4.7.0/lib/rubycritic/commands/default.rb:24:in `critique'
        from /Users/prtngn/.rvm/gems/ruby-3.1.2@telegram_sender/gems/rubycritic-4.7.0/lib/rubycritic/commands/default.rb:19:in `execute'
        from /Users/prtngn/.rvm/gems/ruby-3.1.2@telegram_sender/gems/rubycritic-4.7.0/lib/rubycritic/cli/application.rb:21:in `execute'
        from /Users/prtngn/.rvm/gems/ruby-3.1.2@telegram_sender/gems/rubycritic-4.7.0/bin/rubycritic:10:in `<top (required)>'
        from /Users/prtngn/.rvm/gems/ruby-3.1.2@telegram_sender/bin/rubycritic:25:in `load'
        from /Users/prtngn/.rvm/gems/ruby-3.1.2@telegram_sender/bin/rubycritic:25:in `<main>'
        from /Users/prtngn/.rvm/gems/ruby-3.1.2@telegram_sender/bin/ruby_executable_hooks:22:in `eval'
        from /Users/prtngn/.rvm/gems/ruby-3.1.2@telegram_sender/bin/ruby_executable_hooks:22:in `<main>'
➜  telegram_sender git:(master) ✗   
```

I try to add sexp_processor to Gemfile, but don't working too.

**Comment by nunosilva800** [comment]
The real error is coming from the Flog gem:

```
flog-4.6.5/lib/flog.rb:368:in `process_block_pass': undefined method `sexp_type' for nil:NilClass (NoMethodError)
```

https://github.com/seattlerb/flog/blob/master/lib/flog.rb#L368

A [recent commit](https://github.com/seattlerb/flog/commit/7b774410c127b69b32a29eb11898aa6dfb0121ce) there addressed issues with ruby 3.0. Perhaps this error is due to ruby 3.1.

Do you mind running `flog` on your codebase, perhaps find the file that is causing the problem, and opening an issue there?

**Comment by prtngn** [comment]
Ok
